DNN Platform/DotNetNuke.Abstractions/Portals/IPortalAliasService.cs

```diff
@@ -0,0 +1,99 @@
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Abstractions.Portals
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Portal Alias APIs for managing the different Portal Alias.
+    /// </summary>
+    public interface IPortalAliasService
+    {
+        /// <summary>
+        /// Gets the portal alias by portal.
+        /// </summary>
+        /// <param name="portalId">The portal id.</param>
+        /// <param name="portalAlias">The portal alias.</param>
+        /// <returns>Portal alias.</returns>
+        string GetPortalAliasByPortal(int portalId, string portalAlias);
+
+        /// <summary>
+        /// Gets the portal alias by tab.
+        /// </summary>
+        /// <param name="tabId">The tab ID.</param>
+        /// <param name="portalAlias">The portal alias.</param>
+        /// <returns>Portal alias.</returns>
+        string GetPortalAliasByTab(int tabId, string portalAlias);
+
+        /// <summary>
+        /// Validates the alias.
+        /// </summary>
+        /// <param name="portalAlias">The portal alias.</param>
+        /// <param name="ischild">if set to <c>true</c> [ischild].</param>
+        /// <returns><c>true</c> if the alias is a valid url format; otherwise return <c>false</c>.</returns>
+        bool ValidateAlias(string portalAlias, bool ischild);
+
+        /// <summary>
+        /// Adds a new Portal Alias.
+        /// </summary>
+        /// <param name="portalAlias">The portal alias.</param>
+        /// <returns>The new portal alias ID.</returns>
+        int AddPortalAlias(IPortalAliasInfo portalAlias);
+
+        /// <summary>
+        /// Deletes an existing portal alias.
+        /// </summary>
+        /// <param name="portalAlias">The portal alias to delete.</param>
+        void DeletePortalAlias(IPortalAliasInfo portalAlias);
+
+        /// <summary>
+        /// Get a portal alias by name.
+        /// </summary>
+        /// <param name="alias">The name of the portal alias.</param>
+        /// <returns>The portal alias.</returns>
+        IPortalAliasInfo GetPortalAlias(string alias);
+
+        /// <summary>
+        /// Gets the portal alias.
+        /// </summary>
+        /// <param name="alias">The portal alias.</param>
+        /// <param name="portalId">The portal ID.</param>
+        /// <returns>Portal Alias Info.</returns>
+        IPortalAliasInfo GetPortalAlias(string alias, int portalId);
+
+        /// <summary>
+        /// Gets the portal alias by portal alias ID.
+        /// </summary>
+        /// <param name="portalAliasId">The portal alias ID.</param>
+        /// <returns>Portal alias info.</returns>
+        IPortalAliasInfo GetPortalAliasByPortalAliasID(int portalAliasId);
+
+        /// <summary>
+        /// Get all portal aliases.
+        /// </summary>
+        /// <returns>A collection of portal aliases</returns>
+        PortalAliasCollection GetPortalAliases();
```

Comment:
If we're creating something new, do we want to use a custom collection class?  Or would it be simpler to just use `IDictionary<string, IPortalAliasInfo>`?

Reply:
I agree with you on this, adding our own collection objects is something that I don't see as necessary anymore. I brought it over to keep consistency. I am going to take a 2nd look at the implementation and see if it is truly necessary. I'll report my findings here

Reply:
I would second the desire to rid of the custom collection in favor of a standard collection

Reply:
Also, creating something new, would it be good to have a "GetDefaultPortalAlias" or similar with Portal & Culture?  To fix what people end up needing to do so often

Reply:
Does this look right @mitchelsellers?
```c#
/// <summary>Gets the primary alias for the site.</summary>
/// <param name="portalId">The portal ID.</param>
/// <exception cref="ArgumentException">No portal has the given <paramref name="portalId"/>.</exception>
/// <returns>An <see cref="IPortalAliasInfo" /> instance.</returns>
/// <remarks>If no alias is marked as primary, returns the first neutral culture alias.</remarks>
IPortalAliasInfo GetPrimaryPortalAlias(int portalId);

/// <summary>Gets the primary alias for the site.</summary>
/// <param name="portalId">The portal ID.</param>
/// <param name="cultureCode">The culture code.</param>
/// <exception cref="ArgumentException">No portal has the given <paramref name="portalId"/>, or no alias matches the <paramref name="cultureCode"/>.</exception>
/// <returns>An <see cref="IPortalAliasInfo" /> instance.</returns>
/// <remarks>If no alias is marked as primary, returns the first alias matching the culture.</remarks>
IPortalAliasInfo GetPrimaryPortalAlias(int portalId, string cultureCode);
```

Reply:
@bdukes @mitchelsellers is there any actionable item here that I need to take on the PR?

Reply:
shouldn't the method be named GetPortalAliasByPortalAliasId instead of GetPortalAliasByPortalAliasID?